core/test.lisp

```diff
@@ -37,6 +38,11 @@
                                                          :desc "Raise an error while testing."))
                                   (return nil))))
                  (funcall function)))))
+    (when (and *quit-on-failure*
+               (not (passedp (stats-context *stats*))))
+      (format t "Failed test, with the abort option enabled. ~%")
+      (abort))
```

Comment:
How about sending out some kind of signal and having rove:run / rove:run-test catch it?